src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionKind.cs

```diff
@@ -47,6 +47,11 @@ internal enum ConversionKind : byte
         // IdentityValue is not a part of the language. 
         // It is used by lowering to ensure that trivially reduced expressions 
         // do not become exposed to mutations if used as receivers of struct methods.
-        IdentityValue,  
+        IdentityValue,
+
+        // ObjectToPointer is not directly a part of the language
+        // It is used by lowering of "fixed" statements to represent conversion of an object reference (O) to an unmanaged pointer (*)
+        // The conversion is unsafe and makes sense only if (O) is pinned.
+        ObjectToPointer,
```

Comment:
My understanding is that you're actually converting a pinned ref local to the pointer, not the object itself. Is that not true?